Source/ACE.Server/Network/GameEvent/Events/GameEventFellowshipFullUpdate.cs

```diff
@@ -48,9 +37,14 @@ public GameEventFellowshipFullUpdate(Session session)
 
             Writer.Write(Convert.ToUInt32(fellowship.IsLocked));
 
-            // TODO PackableHashTable of fellows departed - fellowsDeparted  -<ObjectID,int>
-            Writer.Write((uint)0x00200000);
-            Writer.Write((uint)0x00200000);
+            PackableHashTable.WriteHeader(Writer, fellowship.DepartedMembers.Count);
+            foreach (var kvp in fellowship.DepartedMembers)
+            {
+                Writer.Write(kvp.Key);
+                Writer.Write(kvp.Value);
+            }
+
+            Writer.Write(fellowship.FellowshipLocks);
```

Comment:
This does not match https://acemulator.github.io/protocol/Messages/0x02BE-S2C.html, nor does it match any of the code in pdb for acclient.exe

See the relevant functions: UIQueueManager::ProcessNetBlobData -> CM_Fellowship::DispatchUI_FullUpdate -> Fellowship::UnPack
                                                                           -> ClientFellowshipSystem::Handle_Fellowship__FullUpdate
, specifically Fellowship:UnPack

The fellowship data structure ends at DepartedMembers, according to the client processing for this packet

Is this a feature that was added after the client pdb?

Reply:
its not found in UnPack, but is sent across wire. Found in Pack

Reply:
"Is this a feature that was added after the client pdb?"

Any answer to this question?

The pdb client doesn't use this data after DepartedMembers at all. The only explanation would be if this locking data structure was added at the end of the game, after the pdb client.

These PRs also need an explanation of what the changes actually are, the reasoning for the changes, and repro steps to test them. What effect does this have on the client, and how can it be tested?

Reply:
> Not sure where in the client, but aclogview decodes it
it was sent across the wire. it probably has no purpose to the client

Reply:
looking in the final client, sub_5BA630 is Fellowship::UnPack

it looks to have no additional logic, and matches the pdb verison... DepartedFellows is the last thing it uses

if this data is completely unused by anything, and has 0 effect on the client or anything in the game, what is the purpose of this code if it does nothing?

Reply:
to fill out aclogview parse.. the data *was* there as `Writer.Write((uint)0x00200000);`

Reply:
![](https://cdn.discordapp.com/attachments/711116401913364491/725950431360057505/unknown.png)
![](https://cdn.discordapp.com/attachments/711116401913364491/725951903661424650/unknown.png)